assets/js/phoenix_live_view.js

```diff
@@ -2916,6 +2916,7 @@ class ViewHook {
 
   __cleanup__(){
     this.__listeners.forEach(callbackRef => this.removeHandleEvent(callbackRef))
+    delete this.el.phxHookId
```

Comment:
I'm noticing that this is the only time we set attributes on elements directly, and everywhere else we DOM.putPrivate/DOM.private.

I wonder if makes sense to use `DOM.putPrivate` to set the hook id, and then we can remove everything in DOM.private of the view element as part of cleanup?

My thinking is that only ever using the private attribute for metadata like this and cleaning it all up will help prevent this class of bug in the future, but maybe it's more of a refactor than we want to do for this / there's a difference in the phxHookId and other things we store in that private store...